docs/source/json-api/production-setup.rst

```diff
@@ -0,0 +1,300 @@
+For this reason for production setups at a minimum we recommend to use a database
+as a query store, this will allow for more efficient caching of the data to improve
+query performance. Details for enabling a query store are highlighted below.
+
+Query Store
+***********
+
+Query store can be described as a cached search index and is useful for use-cases
+where we need to query large active contract sets. The *HTTP-JSON API* server can be configured
+with PostgreSQL/Oracle(Enterprise Edition only) as the query-store backend.
+
+For example to enable the PostgreSQL backend you can use the ``--query-store-jdbc-config`` flag, as shown below.
+
+.. code-block:: shell
+
+    daml json-api --ledger-host localhost --ledger-port 6865 --http-port 7575 \
+    --query-store-jdbc-config "driver=org.postgresql.Driver,url=jdbc:postgresql://localhost:5432/test?&ssl=true,user=postgres,password=password,start-mode=start-only"
+
+For using the Query Store you'll want your first run to specify ``start-mode=create-only``
+so that all the necessary tables are created. After the first run make
+sure its set to ``start-mode=start-only`` so that it doesn't attempt to create
+the tables again.
+
+If you are not familiar with JDBC URLs, we recommend reading the `PostgreSQL JDBC documentation <https://jdbc.postgresql.org/documentation/head/connect.html>`__
+for more information.
+
+.. note:: The full list of query store configuration flags supported can be seen by running ``daml json-api --help``.
+
+
+Security and privacy
+********************
+
+For an *HTTP-JSON API* server, all data is maintained by the operator of the deployment.
+Thus, it is their responsibility to ensure that the data abides by the necessary
+regulations and confidentiality expectations.
+
+It is recommended to use the tools documented by PostgreSQL to protect data at
+rest and using a secure communication channel between the *HTTP-JSON API* server and the PostgreSQL server.
+
+To protect data in transit and over untrusted networks, the *HTTP-JSON API* server provides
+TLS support, to enable TLS you need to specify the private key for your server and the
+certificate chain via ``daml json-api --pem server.pem --crt server.crt``. You can also
+set a custom root CA certificate used to validate client certificates via ``--cacrt ca.crt``
+
+
+Architecture
+************
+
+Components
+----------
+
+A production setup of the *HTTP-JSON API* will involve the following components:
+
+- the *HTTP-JSON API* server
+- the query store backend database server
+- the ledger
+
+*HTTP-JSON API* server exposes an API to interact with the Ledger and it uses JDBC to interact
+with its underlying query store for caching and serving data efficiently.
+
+The *HTTP-JSON API* server releases are regularly tested with OpenJDK 8 on a 64-bit x86 architecture,
+with Ubuntu 20.04, macOS 11.5.2 and Windows Server 2016.
+
+In production, we recommend running on a 64-bit x86 architecture in a Linux
```

Comment:
```suggestion
In production, we recommend running on a x86_64 architecture in a Linux
```

As far as I know `x86` is the designation for the 32-bit version.

Reply:
I think both here is fine